src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractChangeNamespaceService.cs

```diff
@@ -47,56 +52,81 @@ internal abstract class AbstractChangeNamespaceService<TNamespaceDeclarationSynt
     {
         private static readonly char[] s_dotSeparator = new[] { '.' };
 
+        /// <summary>
+        /// The annotation used to track applicable container in each document to be fixed.
+        /// </summary>
+        protected static SyntaxAnnotation[] ContainerAnnotation { get; } = new[] { new SyntaxAnnotation() };
+
+        protected static SyntaxAnnotation WarningAnnotation { get; }
+            = CodeActions.WarningAnnotation.Create(
+                FeaturesResources.Warning_colon_changing_namespace_may_produce_invalid_code_and_change_code_meaning);
+
         protected abstract TCompilationUnitSyntax ChangeNamespaceDeclaration(
             TCompilationUnitSyntax root, ImmutableArray<string> declaredNamespaceParts, ImmutableArray<string> targetNamespaceParts);
 
         protected abstract SyntaxList<TMemberDeclarationSyntax> GetMemberDeclarationsInContainer(SyntaxNode compilationUnitOrNamespaceDecl);
 
-        protected static SyntaxAnnotation WarningAnnotation { get; }
-            = CodeActions.WarningAnnotation.Create(
-                FeaturesResources.Warning_colon_changing_namespace_may_produce_invalid_code_and_change_code_meaning);
+        protected abstract Task<SyntaxNode> TryGetApplicableContainerFromSpanAsync(Document document, TextSpan span, CancellationToken cancellationToken);
+
+        protected abstract string GetDeclaredNamespace(SyntaxNode container);
+
+        protected abstract Task<ImmutableArray<(DocumentId id, SyntaxNode container)>> CanChangeNamespaceWorkerAsync(Document document, SyntaxNode container, CancellationToken cancellationToken);
+
+        public override async Task<bool> CanChangeNamespaceAsync(Document document, SyntaxNode container, CancellationToken cancellationToken)
+        {
+            var applicableContainers = await CanChangeNamespaceWorkerAsync(document, container, cancellationToken).ConfigureAwait(false);
+            return !applicableContainers.IsDefault;
```

Comment:
that CanChangeNamespaceWorkerAsync seems to be returning something non-boolean indicates to me the name o hte method is wonky.

Reply:
Yes, I agree, I will try to think a better name